src/qiskit_qir/visitor.py

```diff
@@ -86,64 +104,49 @@ def visit_instruction(self, instruction):
 
 
 class BasicQisVisitor(QuantumCircuitElementVisitor):
-    def __init__(self, profile: str = "AdaptiveExecution", kwargs={}):
+    def __init__(self, profile: str = "AdaptiveExecution", **kwargs):
         self._module = None
         self._builder = None
+        self._qis = None
         self._qubit_labels = {}
         self._clbit_labels = {}
         self._profile = profile
         self._capabilities = self._map_profile_to_capabilities(profile)
         self._measured_qubits = {}
-        self._use_static_qubit_alloc = kwargs.get("use_static_qubit_alloc", True)
-        self._use_static_result_alloc = kwargs.get("use_static_result_alloc", True)
         self._emit_barrier_calls = kwargs.get("emit_barrier_calls", False)
         self._record_output = kwargs.get("record_output", True)
 
-    def visit_qiskit_module(self, module):
+    def visit_qiskit_module(self, module: QiskitModule):
         _log.debug(
             f"Visiting Qiskit module '{module.name}' ({module.num_qubits}, {module.num_clbits})"
         )
-        self._module = SimpleModule(
-            name=module.name,
-            num_qubits=module.num_qubits,
-            num_results=module.num_clbits,
+        self._module = module.module
+        context = self._module.context
+        entry = entry_point(
+            self._module, module.name, module.num_qubits, module.num_clbits
         )
-        self._module.use_static_qubit_alloc(self._use_static_qubit_alloc)
-        self._module.use_static_result_alloc(self._use_static_result_alloc)
+        # we update the parent module with the unique name
+        # created when defining in function in case of conflicts
+        module._entry_point = entry.name
```

Comment:
Instead of mutating QiskitModule, why not store the entry point name(s) in BasicQisVisitor itself, then _build_module can ask BasicQisVisitor for it afterwards?